xtask/src/main.rs

```diff
@@ -31,108 +31,74 @@ impl Workspace {
     }
 }
 
-/// A build step for one of the workspaces.
```

Comment:
This is just a preference with no functional change? Just removed the `WorkspaceAction` struct, right?

What happened to `forced_deps`? IIRC that is needed to make `cargo test` from clean work.

Reply:
I am recursively calling `run_action` instead of pushing to `forced_deps`. The removal of `WorkspaceAction` makes it easier to build the `Command` without having to pass around all kinds of fields. Instead the respective method can directly be called. IMO this resulted in much easier to read code.